libsolidity/formal/Z3CHCInterface.h

```diff
@@ -0,0 +1,58 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libsolidity/formal/CHCSolverInterface.h>
+#include <libsolidity/formal/Z3Interface.h>
+
+namespace dev
+{
+namespace solidity
+{
+namespace smt
+{
+
+class Z3CHCInterface: public Z3Interface, public CHCSolverInterface
```

Comment:
Hm... ``Z3Interface`` and ``CHCSolverInterface`` both inherit from ``SolverInterface``... do we need to use virtual inheritance (in ``Z3Interface`` and ``CHCSolverInterface`` that is)? Otherwise ``Z3CHCInterface`` will contain two separate instances of ``SolverInterface``... is that desired or is that a problem?

Reply:
Yes, better use virtual